File: frontend/src/app/sites/scotland-yard/scotland-yard.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {GameSessionService} from "../all/services/http/game-session.service";
import {GameConfig} from "./dto/game/GameConfig";
import {GameState} from "./dto/game/GameState";
import {HeatMapConfig} from "./dto/HeatMapConfig";
import {HeatMapEntry} from "./dto/HeatMapEntry";
import {environment} from "../../../environments/environment";
import {NotificationService} from "../../main/services/notification.service";

@Injectable({
  providedIn: 'root'
})
export class ScotlandYardService extends GameSessionService<GameConfig, GameState> {
  protected readonly baseUri: string = environment.httpApiUrl + '/scotland-yard';

  constructor(httpClient: HttpClient, notification: NotificationService) {
    super(httpClient, notification);
  }

  getHeatMap(heatMapConfig: HeatMapConfig) {
    return this.httpClient.post<HeatMapEntry[]>(`${this.baseUri}/heatmap`, heatMapConfig);
  }
}

File: backend/src/main/java/net/zorphy/backend/site/scotlandyard/service/ScotlandYardService.java
package net.zorphy.backend.site.scotlandyard.service;

import net.zorphy.backend.site.core.http.service.GameSessionService;
import net.zorphy.backend.site.scotlandyard.dto.HeatMapConfig;
import net.zorphy.backend.site.scotlandyard.dto.HeatMapEntry;
import net.zorphy.backend.site.scotlandyard.dto.game.GameConfig;
import net.zorphy.backend.site.scotlandyard.dto.game.GameState;

import java.util.List;

public interface ScotlandYardService extends GameSessionService<GameConfig, GameState> {
    List<HeatMapEntry> computeHeatMap(GameState oldState, HeatMapConfig heatMapConfig);
}
